### PR TITLE
fix: improve subshell detection to avoid false positives with markdown content

### PR DESCRIPTION
**Pull request by daniel-lxs**
Fixes issue where commands containing markdown with backticks were incorrectly denied.

Added intelligent subshell detection that:
- Exempts text output commands (echo, printf, etc)
- Detects markdown patterns to avoid false positives
- Only flags actual command substitution syntax

Solves #6028

----

> [!IMPORTANT]
> Improved subshell detection in `command-validation.ts` to exempt markdown content from blocking, while still blocking actual command substitutions if denylist is present.
> 
>   - **Behavior**:
>     - Improved subshell detection in `command-validation.ts` to avoid false positives with markdown content.
>     - Commands like `echo`, `printf`, and `cat` with markdown are exempt from blocking.
>     - Actual command substitutions are still blocked if denylist is present.
>   - **Functions**:
>     - Added `containsBlockableSubshell()` to intelligently detect subshells.
>     - Updated `isAutoApprovedCommand()`, `isAutoDeniedCommand()`, and `getCommandDecision()` to use new detection logic.
>   - **Tests**:
>     - Added tests in `command-validation.spec.ts` to verify markdown handling and subshell detection.
>     - Tests cover cases with markdown content, actual command substitutions, and edge cases with backticks.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for aebec8b75017144d9fb50a12a2143eb652ae65d4. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: webview-ui/src/utils/command-validation.ts [hunk 1]
```diff
@@ -279,6 +280,40 @@ export function isAutoDeniedSingleCommand(
...
+	if (isTextOutputCommand) return false
+
+	// Look for actual command substitution $()
+	const hasCommandSubstitution = /\$\([^)]+\)/.test(command)
+
+	// For backticks, be more careful - they could be in markdown
+	let hasBacktickSubstitution = false
+	if (command.includes("`")) {
+		// Simple heuristic: if the command has multi-line content or markdown-like
+		// patterns, the backticks are probably not for command substitution
```
Comment (mrubens) [inline webview-ui/src/utils/command-validation.ts:303]: I don’t think a heuristic is enough here. We need to be confident that an attacker who can inject a prompt is not able to execute a command that bypasses the user’s auto-approve configuration.

Comment (daniel-lxs) [reply webview-ui/src/utils/command-validation.ts:303]: I'm not really sure how to fix this issue, basically long commands like the ones used to create PRs with `gh` are failing. 

Also I suspect https://github.com/RooCodeInc/Roo-Code/issues/5997 is caused by a bug in this logic but I wasn't able to reproduce it

Comment (mrubens) [reply webview-ui/src/utils/command-validation.ts:303]: I just fixed https://github.com/RooCodeInc/Roo-Code/pull/6121 this morning, might be related to #5997

Comment (mrubens) [reply webview-ui/src/utils/command-validation.ts:303]: I have an idea for fixing the issue with backticks, will open a separate PR
